### ISSUE TITLE
Async in metrics

### ISSUE THREAD

**Issue by ysard** [labsquare/cutevariant#175]
I tried SqlRunnable for the metrics plugin here:

https://github.com/ysard/cutevariant/blob/metrics/cutevariant/gui/plugins/metrics/dialogs.py

But i didn't succeed to not have a freeze of the interface (for big files ~1.8Go).
The async implementation doesn't provide the expected responsiveness of the interface.

Do you now why ?

Here is the execution flow:

- mainwindow: dialog = dialog_class(conn=self.conn)
- dialogs: self.populate()
- dialogs: "ICI" print => start thread
- dialogs: "CONSTR" => end of the constructor exec
- dialogs: "LA" print => end of the thread

The execution flow is what i expect.
BUT the window is showed few seconds after "ICI" and thus "CONSTR" and "LA" instantly follow each other.
=> The end of __init__ is obtained once the SQL queries are ended.

In conclusion, this seems to not be async. The call to SqlRunnable is blocking here :o

**Comment by dridk** [comment]
Probably because dialog.exec_() start a new event loop.
What if the sql thread start after dialog.exec_() ? 
Or you can try with show instead exec ?

**Comment by ysard** [comment]
- show() didn't show the QDialog at all; even by overriding show() and explicitly call parent func with `super().show()` Nothing is displayed. Don't know why...
- Also tried to override the showEvent: same behavior:
```
    def showEvent(self, event):
        print("show")
        self.model.clear()
        self.populate()
        print("show end")
```
The end of the show event is reached when the thread itself has finished its job.

- Tried with a singleshot timer:
```
    def showEvent(self, event):
        print("show")
        self.model.clear()
        QTimer.singleShot(0, self.populate)
        print("show end")
```
Here i obtain only the edge of the window before obtaining the data:
![image](https://user-images.githubusercontent.com/14988285/96346622-0da8ac80-109d-11eb-8bd9-da9ded00e551.png)
Fun but not expected...

**Comment by dridk** [comment]
I fixed it . But I have no idea how to show my change from github . Don't want to fork your fork .  So i put the diff here . 

You create the pool locally and it has been destroyed at the end of the function. 
You must use the : QThreadPool.globalInstance().start(self.metrics_runnable)

```python
 #  Metrics
@@ -115,13 +116,15 @@ def __init__(self, conn=None, parent=None):
 
         # Async stuff
         self.metrics_thread = None
+        # QTimer.singleShot(1000, self.populate)
         self.populate()
-        print("CONSTR")
 
     def populate(self):
+        print("populate")
 
         def compute_metrics(conn):
-
+            print("start fct", conn)
+            time.sleep(2)
             return {
                 "variants": get_variant_count(conn),
                 "snps": get_snp_count(conn),
@@ -130,20 +133,13 @@ def compute_metrics(conn):
                 "samples": get_sample_count(conn),
             }
 
-        self.metrics_thread = SqlRunnable(
-            self.conn, compute_metrics
-        )
-        self.metrics_thread.finished.connect(self.loaded)
-        pool = QThreadPool()
-        pool.start(self.metrics_thread)
-        print("ICI")
+        self.metrics_runnable = SqlRunnable(self.conn, compute_metrics)
+        QThreadPool.globalInstance().start(self.metrics_runnable)
+        self.metrics_runnable.finished.connect(self.loaded)
 
     def loaded(self):
-        print("LA")
-
-        assert self.metrics_thread.done
 
-        results = self.metrics_thread.results
+        results = self.metrics_runnable.results
 
         self.model.clear()
         ratio = results["transitions"] / results["transversions"]
@@ -169,10 +165,8 @@ def loaded(self):
 
     conn = sql.get_sql_connexion("test.db")
 
-    dialog = MetricsDialog()
-    dialog.conn = conn
-    dialog.populate()
+    dialog = MetricsDialog(conn)
 
-    dialog.show()
+    dialog.exec_()
 
     app.exec_()

```

**Comment by ysard** [comment]
Thank you, I think I would have taken a long time to find the nuance of global instance or I would not have found it.

I will add it to my branch for this time.

About git, it's like preparing a PR on a repository.

```
# Add my repo to yours
git remote add downstream git@github.com:ysard/cutevariant.git
# Now your repo is tied to 2 repos: origin (yours), downstream (mine)
# Sync your git with mine (get the status of existing branches)
git fetch downstream
# go to my branch
git co metrics
# Commit your changes & push to your repo
git push origin metrics
# Now make a PR from github interface
```
